src/components/elements/controlledDisplayFiltersGroup/index.tsx

```diff
@@ -23,28 +23,71 @@ export function ControlledDisplayFiltersGroup({
   label
 }: ControlledDisplayFiltersGroupProps) {
   const [showAll, setShowAll] = useState<boolean>(false);
+  const [displayedFilters, setDisplayedFilters] = useState<Set<string>>(() => new Set<string>());
+
+  const availableFiltersNames= new Set(filters.map((filter)=>filter.label));
 
   const toggleAll = () => {
+    if(!showAll){
+      setDisplayedFilters(availableFiltersNames);
+    }
+    else{
+      setDisplayedFilters(new Set());
+    }
     setShowAll(!showAll);
     toggleShowAll(!showAll);
   };
 
-  const updateFilter = (name: string, show: boolean) => {
-    onFilterChange(name, show);
+  function setsAreEqual(setA:Set<string>, setB:Set<string>) {
+    if (setA.size !== setB.size) {
+      return false;
+    }
+    for (const item of setA) {
+      if (!setB.has(item)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  useEffect(() => {
+    // When metadataToDisplay changes, check if it's equal to availableFiltersNames
+    if (setsAreEqual(displayedFilters, availableFiltersNames)) {
+      setShowAll(true);
+    } else {
+      setShowAll(false);
+    }
+  }, [displayedFilters, availableFiltersNames]);
+
+  const updateFilter = (item: DisplayFilterPartialProps, show: boolean) => {
+    if(item.onChange){
+      item.onChange(show)
+    }
+    onFilterChange(item.label, show);
   };
 
   return (
     <Box sx={{ backgroundColor: 'rgba(248, 250, 252, 1)' }}>
       <Box>
-        <LabeledSwitch label={label} onChange={toggleAll} />
+        <LabeledSwitch label={label} onChange={toggleAll} checked={showAll}/>
```

Comment:
Maybe consider renaming the `label` prop, from looking at the code it's hard to understand the difference between it and the item.label

IIUC, the label prop is the title for the whole group?

Reply:
It the label for the toggle-all switch, I renamed it, so it will be more clear